crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -177,6 +184,17 @@ impl CheckpointExecutor {
                 Some(Ok((checkpoint, checkpoint_execution_state))) = pending.next() => {
                     self.process_executed_checkpoint(&checkpoint, checkpoint_execution_state).await;
                     highest_executed = Some(checkpoint);
+
+                    // Estimate TPS every 10k transactions or 30 sec
+                    let elapsed = now_time.elapsed().as_millis();
+                    let current_transaction_num =  highest_executed.as_ref().map(|c| c.summary.network_total_transactions).unwrap_or(0);
+                    if current_transaction_num - now_transaction_num > 10_000 || elapsed > 30_000{
+                        let tps = (1000.0 * (current_transaction_num - now_transaction_num) as f64 / elapsed as f64) as i32;
+                        self.metrics.checkpoint_exec_sync_tps.set(tps as i64);
```

Comment:
Why not increment a counter and compute the rate later?

Reply:
I am not sure I understand? Is this about computing based on `checkpoint_transaction_count_sum`?

Reply:
I left the comment thinking a counter like `checkpoint_transaction_count_sum` can be used. I see that a gauge for tps is useful here.